.travis.sh

```diff
@@ -37,14 +37,36 @@ install_from_github() {
 }
 
 sudo apt-get update -qq
+
 # libpcsclite-dev is required for OpenSC
 sudo apt-get install -y libpcsclite-dev
 
 export CC=`which $CC`
 mkdir prerequisites
 cd prerequisites
+
+# Install OpenSSL
+if [ -n "${OPENSSL}" ]; then
+    # Remove pre-installed OpenSSL
+    sudo apt-get remove openssl libssl-dev
+
+    OPENSSL_DIR=/usr
+    git clone https://github.com/openssl/openssl.git -b ${OPENSSL}
+    cd "openssl"
+    ./config shared -fPIC --prefix=${OPENSSL_DIR}
+    make depend && make
+    sudo make install
+    cd ..
+    PATH="${OPENSSL_DIR}/bin:${PATH}"
+    CFLAGS="${CFLAGS} -I${OPENSSL_DIR}/include"
+    LDFLAGS="${LD_FLAGS} -L${OPENSSL_DIR}/lib"
+    LD_RUN_PATH="${OPENSSL_DIR}/lib"
+    LD_LIBRARY_PATH="${OPENSSL_DIR}/lib:${LD_LIBRARY_PATH}"
```

Comment:
Sorry for being nit-picky, but changing these variables should not be needed (here and in `.travis.yml`). please just use `sudo ldconfig`.

Reply:
I too wanted to get rid of these exports. Simply calling ``sudo ldconfig`` did not solve the issue. But following [this guideline](https://wiki.openssl.org/index.php/Compilation_and_Installation#PREFIX_and_OPENSSLDIR) to set the flags for both OpenSSL 1.0.2 and OpenSSL 1.1.0, and changing the prefix for OpenSSL helped.

The final issue was SoftHSM configuration not finding OpenSSL. Setting manually the prefix using ``--with-openssl`` did the trick.